File: python/task1b/S1_IF_03.py
# S1_IF_03 - Normalize user name

def normalize_name(x):
    if not x:
        return "Anonymous"
    if not isinstance(x, str):
        return "Anonymous"

    stripped = x.strip()
    if not stripped:
        return "Anonymous"

    return stripped

# Tests
tests = ["", " ", None, " Ola ", "  Bob  ", 123, "Alice"]

print(f"{'input':<12} {'result'}")
print("-" * 28)
for t in tests:
    print(f"{repr(t):<12} '{normalize_name(t)}'")

File: js/task1b/S1_IF_03.js
// S1_IF_03 - Truthy/falsy guard

function normalizeName(input) {
  if (!input) return "Anonymous";
  return input.trim();
}

// tests
console.log('normalizeName(""):', normalizeName(""));           // Anonymous
console.log('normalizeName(" "):', normalizeName(" "));         // "" (empty after trim)
console.log('normalizeName(null):', normalizeName(null));       // Anonymous
console.log('normalizeName(" Ola "):', normalizeName(" Ola ")); // "Ola"
console.log('normalizeName(undefined):', normalizeName(undefined)); // Anonymous
console.log('normalizeName(0):', normalizeName(0));             // Anonymous (0 is falsy)
